plugins/plugin-chart-table/src/TableChart.tsx

```diff
@@ -161,13 +161,51 @@ export default function TableChart<D extends DataRecord = DataRecord>(
     showCellBars = true,
     emitFilter = false,
     sortDesc = false,
-    onChangeFilter,
-    filters: initialFilters,
+    filters: initialFilters = {},
     sticky = true, // whether to use sticky header
   } = props;
 
   const [filters, setFilters] = useState(initialFilters);
 
+  const handleChange = useCallback(
+    (filters: { [x: string]: DataRecordValue[] }) => {
+      if (!emitFilter) {
+        return;
+      }
+
+      const groupBy = Object.keys(filters);
+      const groupByValues = Object.values(filters);
+      setDataMask({
+        crossFilters: {
+          extraFormData: {
+            append_form_data: {
```

Comment:
I'm wondering whether there is room for a simpler public API to set cross filters. Now It feels a little tedious that every chart with corss-filter support has to set this very long nested object themselves.

Reply:
We're actually currently cleaning up the API (some of this is leftovers from during development when it wasn't clear in which direction this was going to ultimately go)